### PR TITLE
SW-2608 Add IDs to some components to allow for easier tracking

### PR DESCRIPTION
**Pull request by alexTerraformation**
In GTM we have some triggers on appearance of error text labels.  When reporting which label appeared it is useful to have a value in the "for" attribute.

### CODE REVIEW

File: src/components/accession2/properties/StorageSubLocationSelector.tsx [hunk 1]
```diff
@@ -3,6 +3,7 @@ import strings from 'src/strings';
 import { Select } from '@terraware/web-components';
 
 type StorageSubLocationSelectorProps = {
+  id?: string;
```
Comment (karthikbtf) [inline src/components/accession2/properties/StorageSubLocationSelector.tsx:6]: use the id below?

File: src/components/accession2/edit/EditLocationModal.tsx [hunk 2]
```diff
@@ -89,6 +89,7 @@ export default function EditLocationModal(props: EditLocationModalProps): JSX.El
       <Grid item xs={12} textAlign='left'>
         <Grid item xs={12}>
           <StorageLocationSelector
+            id='location'
```
Comment (karthikbtf) [inline src/components/accession2/edit/EditLocationModal.tsx:92]: not sure if you want to distinguish between this use-case and the other.. if so, maybe call it `edit-location` .. etc.

**Review by karthikbtf — APPROVED** [review]
🎉 thanks.. some comments

File: src/components/accession2/edit/EditLocationModal.tsx (hunk 2 above)
Comment (karthikbtf) [reply src/components/accession2/edit/EditLocationModal.tsx:92]: i guess the page url helps distinguish here,, since these flows aren't modals

Comment (alexTerraformation) [reply src/components/accession2/edit/EditLocationModal.tsx]: Good practice anyways!

**Review by karthikbtf — APPROVED** [review]

**Pull request merged**